packages/react/src/components/UIShell/SideNavButton.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright IBM Corp. 2025,
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import PropTypes from 'prop-types';
+import React, {
+  ElementType,
+  ForwardedRef,
+  useContext,
+  Ref,
+  ComponentProps,
+} from 'react';
+import Button from '../Button';
+import SideNavItem from './SideNavItem';
+import cx from 'classnames';
+import { SideNavContext } from './SideNav';
+
+export type SideNavButtonProps = ComponentProps<typeof Button> & {
+  /**
+   * Specify the children to be rendered inside of the `SideNavMenuItem`
+   */
+  children?: React.ReactNode;
+
+  /**
+   * Provide an optional class to be applied to the containing node
+   */
+  className?: string;
+
+  /**
+   * Property to indicate if the side nav container is open (or not). Use to
+   * keep local state and styling in step with the SideNav expansion state.
+   */
+  isSideNavExpanded?: boolean;
+
+  /**
+   * Optional prop to specify the tabIndex of the button. If undefined, it will be applied default validation
+   */
+  tabIndex?: number;
+};
+
+const SideNavButton = React.forwardRef<HTMLButtonElement, SideNavButtonProps>(
+  function SideNavButton(props, ref: ForwardedRef<HTMLButtonElement>) {
```

Comment:
- `ref` is already typed.
- An anonymous function can be used. `SideNavButton` is never gonna be called.

```suggestion
  (props, ref) => {
```